client/main-layout.tsx

```diff
@@ -0,0 +1,401 @@
+import Whisper from './whispers/whisper'
+
+const curVersion = __WEBPACK_ENV.VERSION
+
+const KEY_C = keycode('c')
+const KEY_D = keycode('d')
+const KEY_F = keycode('f')
+const KEY_J = keycode('j')
+const KEY_M = keycode('m')
+const KEY_R = keycode('r')
+
+const Container = styled.div`
+  display: flex;
+  overflow: hidden;
+`
+
+const Content = styled.div`
+  flex-grow: 1;
+  flex-shrink: 1;
+  overflow-x: hidden;
+`
+
+const StyledMapsIcon = styled(MapsIcon)`
+  width: 36px;
+  height: 36px;
+`
+
+const VersionText = styled.div`
+  ${caption};
+  margin: 8px 0px 0px 0px;
+  color: ${colorTextSecondary};
+  letter-spacing: 1.25px;
+`
+
+let lobbyRoute = <></>
+let matchmakingRoute = <></>
+let partyRoute = <></>
+if (IS_ELECTRON) {
+  // TODO(2Pac): Remove `any` once the `LobbyView` is TS-ified
+  lobbyRoute = <Route path='/lobbies/:lobby/:rest*' component={LobbyView as any} />
+  matchmakingRoute = <Route path='/matchmaking/:rest*' component={MatchmakingView} />
+  partyRoute = <Route path='/parties/:partyId/:rest*' component={PartyView} />
+}
+
+const LoadableAdminPanel = loadable(() => import('./admin/panel'), {
+  // TODO(tec27): do we need to position this indicator differently? (or pull that into a common
+  // place?)
+  fallback: <LoadingIndicator />,
+})
+
+const MiniActivityButtonsContainer = styled.div`
+  width: 100%;
+  display: flex;
+`
+
+const FadedSettingsIcon = styled(SettingsIcon)`
+  color: ${colorTextSecondary};
+`
+
+/**
+ * Tracks if this is the first time this user has logged in on this client. Pretty dumb, if we need
+ * more smarts we can add it as a Context var or put it in the store or something.
+ */
+let firstLoggedIn = true
+
+export function MainLayout() {
+  const dispatch = useAppDispatch()
+  const inGameplayActivity = useAppSelector(s => s.gameplayActivity.inGameplayActivity)
+  const isEmailVerified = useAppSelector(s => s.auth.user.emailVerified)
+  const isMatchmakingSearching = useAppSelector(s => !!s.matchmaking.searchInfo)
+  const lobbyCount = useAppSelector(s => s.serverStatus.lobbyCount)
+  const starcraft = useAppSelector(s => s.starcraft)
+
+  const [searchingMatchOverlayOpen, setSearchingMatchOverlayOpen] = useState(false)
+
+  const findMatchButtonRef = useRef<HTMLButtonElement>()
+  const searchingMatchButtonRef = useRef<HTMLButtonElement>()
+
+  useEffect(() => {
+    dispatch(openChangelogIfNecessary())
+
+    if (firstLoggedIn) {
+      firstLoggedIn = false
+      if (!isEmailVerified) {
+        dispatch(
+          addLocalNotification({
+            id: EMAIL_VERIFICATION_ID,
+            type: NotificationType.EmailVerification,
+          }),
+        )
+      }
+
+      dispatch(addPrivacyPolicyNotificationIfNeeded())
+      dispatch(addTermsOfServiceNotificationIfNeeded())
+      dispatch(addAcceptableUseNotificationIfNeeded())
+    }
+
+    return () => {
+      firstLoggedIn = true
+    }
+  }, [isEmailVerified, dispatch])
+
+  const onFindMatchClick = () => {
```

Comment:
So I know we talked about how we're probably overusing `useCallback`, *but*, when you're updating existing code, especially the code that renders basically everything else in the app, you should probably try to maintain the existing semantics as much as possible. 

If you can absolutely verify that having stable references for these isn't necessary, sure, you can make that change. But it doesn't really seem like that's the case here, so we should probably leave them as-is (that is, wrap these in `useCallback`).

For new code this decision often has less consequences, but for stuff like this I'd focus on changing as little as possible unless you know you have covered all the permutations (ideally with tests, but at the very least with manual testing)